### PR TITLE
Homework Podstawy C++ #2

### PR DESCRIPTION
**Pull request by niemarcin**
Authors: @niemarcin, @lisie-sprawy

- [x] Zadanie 1 - AddEven  **- done**
- [x] Zadanie 2 - NWD (LCM) i NWW (GCD) **- done**
- [x] Zadanie 3 - MaxOfVector **- done**
- [x] Zadanie 4 - GenerateSequence **- done**

Accidentally tasks from the last lesson are also added here.

### CODE REVIEW

File: module2/homework/addEven/addEven.hpp [hunk 1]
```diff
@@ -2,7 +2,12 @@
...
 
 int addEven(const std::vector<int>& numbers) {
-    // TODO: Your implementation goes here
-    // Below return is only to make this function compile now
-    return -1;
+    int suma = 0;
+
+    for (const auto & element : numbers){
+        if (element % 2 == 0) {
+        suma = suma + element;
```
Comment (jakjan95) [inline module2/homework/addEven/addEven.hpp:9]: You can use `+=` operator here

File: module2/homework/maxOfVector/maxOfVector.hpp [hunk 2]
```diff
@@ -3,6 +3,12 @@
 #include <limits>
 
 int maxOfVector(const std::vector<int>& vec) {
-    // TODO: Implement me :)
-    return {};
+    int maxValue = INT32_MIN;
+    for (const auto element : vec) {
```
Comment (jakjan95) [inline module2/homework/maxOfVector/maxOfVector.hpp:7]: If you want refer to orginal item without copying it you need to use const reference which is `const auto&`

File: module2/homework/addEven/addEven.hpp [hunk 3]
```diff
@@ -2,7 +2,12 @@
 #include <vector>
 
 int addEven(const std::vector<int>& numbers) {
-    // TODO: Your implementation goes here
-    // Below return is only to make this function compile now
-    return -1;
+    int suma = 0;
+
+    for (const auto & element : numbers){
+        if (element % 2 == 0) {
```
Comment (DamiXioR) [inline module2/homework/addEven/addEven.hpp:8]: i think code would look better with (!(element % 2))

File: module2/homework/addEven/addEven.hpp [hunk 4]
```diff
@@ -2,7 +2,12 @@
 #include <vector>
 
 int addEven(const std::vector<int>& numbers) {
-    // TODO: Your implementation goes here
-    // Below return is only to make this function compile now
-    return -1;
+    int suma = 0;
```
Comment (piezol) [inline module2/homework/addEven/addEven.hpp:5]: Try to use only english in your code.

File: module2/homework/nwdNww/nwdNww.hpp [hunk 5]
```diff
@@ -1,11 +1,15 @@
...
+    return std::abs(lhs); 
 }
 
 int NWW(int lhs, int rhs) {
-    // TODO: Implement me :)
-    return -1;
+    if (lhs==0 || rhs==0){
+        return 0;
+    }
+    return std::abs(lhs * rhs) / NWD(lhs,rhs);
```
Comment (piezol) [inline module2/homework/nwdNww/nwdNww.hpp:14]: I'm copying my comment from different PR:
There are trivial cases that your code won't work for, ie NWW(1e8, 1e8).
That is due to lhs*rhs overflowing int.
You can fix it by casting lhs to unsigned long long in the multiplication.

There are some nontrivial cases that will still fail, ie NWW(1e8-1, 1e8+1), but they fail for the std:lcm(int, int) as well, so it's ok.

File: module2/task1.cpp [hunk 6]
```diff
@@ -2,8 +2,12 @@
 #include <vector>
 #include <string>
 
-// Implement printVec to print the content of a given vector on screen.
-// Each string in a new line.
+void printVector(std::vector<std::string> vectorToPrint) {
```
Comment (piezol) [inline module2/task1.cpp:5]: Task*cpp files are not a part of our homework.

File: module2/homework/maxOfVector/maxOfVector.hpp (hunk 2 above)
Comment (niemarcin) [reply module2/homework/maxOfVector/maxOfVector.hpp]: Thanks, it's done now.

File: module2/homework/addEven/addEven.hpp (hunk 3 above)
Comment (niemarcin) [reply module2/homework/addEven/addEven.hpp]: Thanks, it's done now.

File: module2/homework/addEven/addEven.hpp (hunk 4 above)
Comment (niemarcin) [reply module2/homework/addEven/addEven.hpp]: Thanks, it's done now.

**Comment by ziobron** [comment]
+4 XP addEven
+6 XP NwwNwd
+5 XP maxOfVector
+5 XP generateSequence
+8 XP delivery before 31.05
+3 XP cooperation

{
  "requirement_ids": [1119, 1631, 1119, 1633, 1119, 1634, 1119, 1635, 1636],
  "points": [2, 4, 2, 6, 2, 5, 2, 5, 3],
  "users": ["niemarcin", "lisie-sprawy"]
}

**Pull request closed without merging**